sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -56,6 +57,11 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
             }
         }
 
+#if NET6_0_OR_GREATER
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(HttpMessage))]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The values being passed into Write have the commonly used properties being preserved with DynamicDependency.")]
```

Comment:
When suppressing these warnings, we typically scope the suppression to just the line, or lines, that are causing the warning. Suppressing a whole method has the risk of new code getting added that is no longer safe, but the suppression doesn't let the warning occur.

Reply:
Since you can't put an attribute on a statement we typically extract the problematic call into a location function and put the suppression on the local function.

Reply:
Will update this in all relevant places.